contracts/contracts/gatekeepers/EASGatekeeper.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+import { SignUpGatekeeper } from "./SignUpGatekeeper.sol";
+import { IEAS } from "../interfaces/IEAS.sol";
+
+/// @title EASGatekeeper
+/// @notice A gatekeeper contract which allows users to sign up to MACI
+/// only if they own an attestation from a valid schema and trusted attester.
+contract EASGatekeeper is SignUpGatekeeper, Ownable {
+  // the reference to the EAS contract
+  IEAS private immutable eas;
+
+  // the schema to check against
+  bytes32 public immutable schema;
+
+  // the trusted attester
+  address public immutable attester;
+
+  /// @notice the reference to the MACI contract
+  address public maci;
+
+  // a mapping of attestations that have already registered
+  mapping(bytes32 => bool) public registeredAttestations;
+
+  /// @notice custom errors
+  error AttestationRevoked();
+  error AlreadyRegistered();
+  error AttesterNotTrusted();
+  error NotYourAttestation();
+  error InvalidSchema();
+  error OnlyMACI();
+  error ZeroAddress();
+
+  /// @notice Deploy an instance of EASGatekeeper
+  /// @param _eas The EAS contract
+  /// @param _attester The trusted attester
+  /// @param _schema The schema UID
```

Comment:
Should this be `schema UID` or `attestation ID`?

Reply:
attester is the address that created the attestation. Maybe we can change schema name to chemaId but sounds kinda understandable either way?